thumbor/doctor.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# thumbor imaging service
+# https://github.com/thumbor/thumbor/wiki
+
+# Licensed under the MIT license:
+# http://www.opensource.org/licenses/mit-license
+# Copyright (c) 2011 globo.com thumbor@googlegroups.com
+
+import argparse
+import sys
+from importlib import import_module
+
+import colorful as cf
+
+from thumbor import __release_date__, __version__
+from thumbor.ext import BUILTIN_EXTENSIONS
+from thumbor.filters import BUILTIN_FILTERS
+
+
+def get_options():
+    parser = argparse.ArgumentParser(description="thumbor doctor")
+
+    parser.add_argument(
+        "-n",
+        "--nocolor",
+        action="store_true",
+        help="Disables coloring of thumbor doctor",
+    )
+
+    options = parser.parse_args()
+
+    return {
+        "nocolor": options.nocolor,
+    }
+
+
+def header(msg, color=cf.yellow):
+    print(color(msg))
+
+
+def subheader(msg, color=cf.bold_coral):
+    print(color(msg))
+    newline()
+
+
+def newline():
+    print()
+
+
+def check_filters():
+    newline()
+    subheader('Verifying thumbor filters...')
+    errors = []
+
+    for filter_name in BUILTIN_FILTERS:
+        try:
+            import_module(filter_name)
+            print(cf.bold_green('✅ %s' % filter_name))
+        except ImportError as error:
+            print(cf.bold_red('❎ %s' % filter_name))
+            errors.append(error)
+
+    return errors
+
+
+def check_extensions():
+    newline()
+    subheader('Verifying thumbor compiled extensions...')
+    errors = []
+
+    for extension in BUILTIN_EXTENSIONS:
+        ext_name = extension.replace('thumbor.ext.filters.', '')
+        try:
+            import_module(extension)
+            print(cf.bold_green('✅ %s' % ext_name))
+        except ImportError as error:
+            print(cf.bold_red('❎ %s' % ext_name))
+            errors.append(error)
+
+    return errors
+
+
+def check_modules():
+    newline()
+    subheader('Verifying pycurl support...')
+    errors = []
+
+    modules = (
+        ('pycurl', 'Thumbor works much better with PyCurl. For more information visit http://pycurl.io/.'),
+        ('cv2', 'Thumbor requires OpenCV for smart cropping. For more information check https://opencv.org/.'),
```

Comment:
should also include exiv2 and cairosvg?

Reply:
Also added checks for programs (jpegtran, gifsicle and ffmpeg).

Reply:
aren't ptah to ffmpeg (at least) has to be configured in config var? In that case, can't rely on `which`

Reply:
It may be added in config, but usually not being able to `which` it is a good indicator that it's not found. I plan on adding a `-c config_path` so we can get more stuff from the config and test it (are paths to storage readable and writable?). But for now I think this is much better than nothing :D